src/middlewares/verifySearchKeys.js

```diff
@@ -0,0 +1,31 @@
+import responseHelper from '../utils/responseHelper';
+
+export default (req, res, next) => {
+  const { query } = req;
+  const keys = Object.keys(query);
+  if (keys.length === 0) {
+    return responseHelper(res, 'No search Parameters Provided', null, 404);
+  }
+
+  const searchKeys = keys.map(key => {
+    if (key === 'location'
+      || key === 'name'
+      || key === 'description'
+      || key === 'highlights'
+      || key === 'amenities') {
+      return true;
+    }
+    return false;
+  });
+
+  const invalidKeys = searchKeys.filter(key => key === false);
+  if (invalidKeys.length > 0) {
+    return responseHelper(res, 'You Provided Invalid Search Keys. Only Name, Description & Locaton Allowed.', null, 400);
```

Comment:
Kindly update the `validation error message` to also include `amenities` and `highlights`.